sqllogictest-bin/src/engines.rs

```diff
@@ -113,6 +115,22 @@ impl AsyncDB for Engines {
     type ColumnType = DefaultColumnType;
 
     async fn run(&mut self, sql: &str) -> Result<DBOutput<Self::ColumnType>, Self::Error> {
-        self.run(sql).await.map_err(EnginesError)
+        dispatch_engines!(self, e, {
+            e.run(sql)
+                .await
+                .map_err(|e| EnginesError(anyhow::Error::from(e)))
+        })
+    }
+
+    fn engine_name(&self) -> &str {
+        dispatch_engines!(self, e, { e.engine_name() })
+    }
+
+    async fn sleep(dur: Duration) {
+        tokio::time::sleep(dur).await
+    }
+
+    async fn run_command(command: std::process::Command) -> std::io::Result<ExitStatus> {
+        Command::from(command).status().await
     }
```

Comment:
This is a breaking api change. Is it possible to make them default methods?

Reply:
It does have default implementations. However, new variants in `RecordOutput` and `TestErrorKind` still break the API. I guess we have to bump the major version.